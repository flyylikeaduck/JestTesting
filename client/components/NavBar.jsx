import React from 'react';
import { Navbar } from 'react-bootstrap';

export default () => (
  <Navbar collapseOnSelect>
    <Navbar.Header>
      <Navbar.Brand>
        <a href="/">
          Jest Testing
        </a>
      </Navbar.Brand>
    </Navbar.Header>
  </Navbar>
);
