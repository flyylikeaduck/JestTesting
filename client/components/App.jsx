import React from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import {
  Grid, Row, Col,
  Button,
} from 'react-bootstrap';

import NavBar from './NavBar';

import { incrementCounter } from '../store/modules/counter';

export const App = ({ count, buttonCallback }) => (
  <div>
    <NavBar />
    <Grid>
      <Row>
        <Col md={1} mdOffset={5}>
          <Button onClick={buttonCallback}>Click Me</Button>
        </Col>
        <Col md={1}>
          Count: {count}
        </Col>
      </Row>
    </Grid>
  </div>
);

App.propTypes = {
  count: PropTypes.number.isRequired,
  buttonCallback: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  count: state.counter.count,
});

const mapDispatchToProps = dispatch => bindActionCreators({
  buttonCallback: incrementCounter,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(App);
