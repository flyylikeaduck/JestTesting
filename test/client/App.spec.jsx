import React from 'react';
import { Button } from 'react-bootstrap';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { App } from '../../client/components/App';

describe('<App />', () => {
  let wrapper;
  let fn;

  beforeEach(() => {
    fn = jest.fn();
    wrapper = shallow(<App count={0} buttonCallback={fn} />);
  });

  it('should render', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call the callback on click', () => {
    wrapper.find(Button).simulate('click');

    expect(fn).toBeCalled();
  });
});
