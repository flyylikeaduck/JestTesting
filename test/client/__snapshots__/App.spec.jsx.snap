// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render 1`] = `
<div>
  <Component />
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={false}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        md={1}
        mdOffset={5}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          onClick={[Function]}
        >
          Click Me
        </Button>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        md={1}
      >
        Count: 
        0
      </Col>
    </Row>
  </Grid>
</div>
`;
